index.js

```diff
@@ -51,7 +53,10 @@ module.exports = function (str, opts) {
 	}
 
 	var ret = lines.map(function (line, lineIndex) {
-		var ret = {};
+		var ret = headers.reduce(function (ret, header) {
+			ret[header] = '';
+			return ret;
+		}, {});
```

Comment:
Fixed [here](https://github.com/dbkaplun/parse-columns/commit/8d704135f0aa1058d355504110ca1bab60e8f757#diff-168726dbe96b3ce427e7fedce31bb0bcL54). There is no need for two separate loops, as we can use the same index for `headers` as for `line`. Also this way `opts.transform` is supported as well.